### PR TITLE
Added a function which checks that all values in the 'lateness' column are numbers

### PR DESCRIPTION
**Pull request by Annie-EXE**

### CODE REVIEW

File: pipeline/test_transform.py [hunk 1]
```diff
@@ -4,7 +4,8 @@
 import pytest
```
Comment (NgawangT) [inline pipeline/test_transform.py:4]: A doc string would be nice.

File: pipeline/test_transform.py [hunk 2]
```diff
@@ -81,3 +82,6 @@ def test_create_timestamp_generates_correct_values():
     datetime_value = df.loc[0, "datetime"]
     assert datetime_value == pd.Timestamp("2023-09-05 12:30:45")
     assert isinstance(datetime_value, pd.Timestamp)
+
+
+def test_
```
Comment (NgawangT) [inline pipeline/test_transform.py:87]: ?

File: pipeline/test_transform.py [hunk 3]
```diff
@@ -4,7 +4,8 @@
 import pytest
 
 from transform import (
-    create_timestamp_from_date_and_time
+    create_timestamp_from_date_and_time,
+    replace_non_integers_with_none
 )
```
Comment (NgawangT) [inline pipeline/test_transform.py:9]: Good import conventions follows single line comments.

File: pipeline/transform.py [hunk 4]
```diff
@@ -65,23 +65,37 @@ def create_timestamp_from_date_and_time(df: DataFrame, new_column_name: str,
     return df
```
Comment (NgawangT) [inline pipeline/transform.py:65]: ?

File: pipeline/transform.py [hunk 5]
```diff
@@ -65,23 +65,37 @@ def create_timestamp_from_date_and_time(df: DataFrame, new_column_name: str,
...
                                                      "scheduled_arrival_date",
                                                      "scheduled_arrival_time")
-
+    
     service_df = create_timestamp_from_date_and_time(service_df,
                                                      "origin_run_datetime",
                                                      "origin_run_date",
                                                      "origin_run_time")
+    
+    # Functions to replace a lateness value with None if the service was cancelled at origin
```
Comment (NgawangT) [inline pipeline/transform.py:94]: better to take this comment out and instead have a doc string in the function itself.

File: pipeline/transform.py [hunk 6]
```diff
@@ -65,23 +65,37 @@ def create_timestamp_from_date_and_time(df: DataFrame, new_column_name: str,
...
+    service_df = replace_non_integers_with_none(service_df, "arrival_lateness")
+    
+    print(service_df)
```
Comment (NgawangT) [inline pipeline/transform.py:97]: do we need this print or was it just for testing purposes?

File: pipeline/transform.py [hunk 7]
```diff
@@ -65,23 +65,37 @@ def create_timestamp_from_date_and_time(df: DataFrame, new_column_name: str,
...
+
 
     output_csv_path = "transformed_service_data.csv"
```
Comment (NgawangT) [inline pipeline/transform.py:100]: Making this a constant at the top of the file OUTPUT_CSV_PATH would be nice.

File: pipeline/test_transform.py (hunk 2 above)
Comment (Annie-EXE) [reply pipeline/test_transform.py:87]: I shall complete the test

File: pipeline/transform.py (hunk 6 above)
Comment (Annie-EXE) [reply pipeline/transform.py:97]: just for testing! I'll take it out

File: pipeline/test_transform.py (hunk 2 above)
Comment (LisheinLWM) [reply pipeline/test_transform.py:87]: You can expand the file up and down :)

File: pipeline/transform.py (hunk 4 above)
Comment (LisheinLWM) [reply pipeline/transform.py:65]: You can expand the file up and down :)

File: pipeline/transform.py [hunk 8]
```diff
@@ -65,23 +65,37 @@ def create_timestamp_from_date_and_time(df: DataFrame, new_column_name: str,
     return df
 
 
+def replace_non_integers_with_none(df: DataFrame, column_name: str) -> DataFrame:
+
+    df[column_name] = df[column_name].apply(lambda x: int(x)
+                                            if str(x).lstrip('-').isdigit()
+                                            else None)
+
+
```
Comment (LisheinLWM) [inline pipeline/transform.py:74]: No need for 2 empty lines here

**Pull request merged**